rllib/core/rl_module/rl_module.py

```diff
@@ -111,6 +113,16 @@ def from_dict(cls, d):
             catalog_class=catalog_class,
         )
 
+    def update(self, other):
+        """Updates this spec with the given other spec. Works like dict.update()."""
+        assert type(other) is SingleAgentRLModuleSpec
```

Comment:
Never use assert for actual errors. Do this instead:
```
if not isisntance(other, Single):
     raise ValueError(...)
```

Reply:
Changed, however, I don't understand why. An AssertionError implicitly what's wrong in the same line, with more info in the error message. I'm always happy when I get them! 😄 
So what's better with the ValueError?

Reply:
I don't know. Someone thought me in grad school to not use assertion since people are lazy and won't put the descriptive message.